progress/internal/linear-progress.ts

```diff
@@ -28,16 +29,20 @@ export class LinearProgress extends Progress {
         (this.indeterminate ? 1 : this.value / this.max) * 100
       }%)`,
     };
+
+    const bufferValue = this.buffer ?? 0;
+    const hasBuffer = bufferValue > 0;
+
+    const dotSize = this.indeterminate || !hasBuffer ? 1 : bufferValue / this.max;
+
     const dotStyles = {
-      transform: `scaleX(${
-        (this.indeterminate ? 1 : this.buffer / this.max) * 100
-      }%)`,
+      transform: `scaleX(${dotSize * 100}%)`,
     };
 
     // Only display dots when visible - this prevents invisible infinite
     // animation.
     const hideDots =
-      this.indeterminate || this.buffer >= this.max || this.value >= this.max;
+      this.indeterminate || !hasBuffer || this.value >= this.max;
```

Comment:
Last change I think! We still need the max check, `!hasBuffer || bufferValue >= this.max`

Reply:
Oops 😬 sorry I forgot about that one after removing the previous one. It should be updated now.